### PR TITLE
feat(19613): refactor authentication factory

### PR DESCRIPTION
**Pull request by nnorbert**
refactor authentication factory

### CODE REVIEW

File: apps/payments/app/[locale]/(hosted)/logto_integration/page.tsx
Comment (SamSalvatico) [inline apps/payments/app/[locale]/(hosted)/logto_integration/page.tsx:1]: I think we can remove this test page

**Review by SamSalvatico — APPROVED** [review]
LGTM

File: apps/payments/app/[locale]/(hosted)/logto_integration/page.tsx
Comment (nnorbert) [reply apps/payments/app/[locale]/(hosted)/logto_integration/page.tsx:1]: Alfonso ask to leave the page there for debugging. Until we are not in production, it is not a problem.

**Review by SamSalvatico — APPROVED** [review]

**Review by peschina — APPROVED** [review]
LGTM

**Pull request merged**